front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/StorageServiceSupport.java

```diff
@@ -78,6 +78,24 @@ public StorageServiceSupport(ObjectType objectType,
     this.shouldWarmCache = shouldWarmCache;
     this.registry = registry;
 
+    // cache object keys for at least two refresh intervals
+    // (will ensure that concurrent requests are debounced across a refresh cycle)
+    long cacheExpiry = refreshIntervalMs * 2;
+    log.debug("Creating object keys cache (expiry: {} minutes)", TimeUnit.MILLISECONDS.toMinutes(cacheExpiry));
+    this.objectKeysByLastModifiedCache = CacheBuilder
+      .newBuilder()
+      .expireAfterWrite(cacheExpiry, TimeUnit.MILLISECONDS)
+      .recordStats()
```

Comment:
There is a loading cache metric instrumenter in clouddriver-aws if you think any of these are worth graphing

Reply:
Thanks for the pointer, I think the logs will be sufficient initially but I might flip on the instrumenter afterwards.